packages/transport/src/modules/relationshipTemplates/local/SendRelationshipTemplateParameters.ts

```diff
@@ -27,7 +30,20 @@ export class SendRelationshipTemplateParameters extends Serializable implements
     @serialize()
     public forIdentity?: CoreAddress;
 
+    @validate({ nullable: true, disallowedValues: [""] })
+    @serialize()
+    public password?: string;
+
+    @validate({ nullable: true, regExp: /^(pw|pin(4|5|6|7|8|9|10|11|12|13|14|15|16))$/ })
+    @serialize()
+    public passwordType?: string;
+
     public static from(value: ISendRelationshipTemplateParameters): SendRelationshipTemplateParameters {
-        return this.fromAny(value);
+        const parameters = this.fromAny(value);
+        if (!parameters.password !== !parameters.passwordType) {
+            throw TransportCoreErrors.general.onlyOneOfPasswordTypeAndPasswordSet();
+        }
```

Comment:
also here - wrong place for the validation + please use serval validation errors instead of core errors for validation issues here.